vm/src/bench/kernel.rs

```diff
@@ -0,0 +1,421 @@
+use cid::Cid;
+use fvm::call_manager::CallManager;
+use fvm::gas::{Gas, GasTimer, PriceList};
+use fvm::kernel::{
+    ActorOps, BlockId, BlockRegistry, BlockStat, CircSupplyOps, CryptoOps, DebugOps, EventOps,
+    ExecutionError, GasOps, IpldBlockOps, LimiterOps, MessageOps, NetworkOps, RandomnessOps,
+    SelfOps, SendResult,
+};
+use fvm::machine::limiter::MemoryLimiter;
+
+use fvm::{DefaultKernel, Kernel};
+use fvm_shared::address::{Address, SECP_PUB_LEN};
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::consensus::ConsensusFault;
+use fvm_shared::crypto::signature::{SignatureType, SECP_SIG_LEN, SECP_SIG_MESSAGE_HASH_SIZE};
+use fvm_shared::econ::TokenAmount;
+
+use fvm_shared::piece::PieceInfo;
+use fvm_shared::randomness::RANDOMNESS_LENGTH;
+use fvm_shared::sector::{
+    AggregateSealVerifyProofAndInfos, RegisteredSealProof, ReplicaUpdateInfo, SealVerifyInfo,
+    WindowPoStVerifyInfo,
+};
+use fvm_shared::sys::out::network::NetworkContext;
+use fvm_shared::sys::out::vm::MessageContext;
+use fvm_shared::sys::SendFlags;
+use fvm_shared::{ActorID, MethodNum, TOTAL_FILECOIN};
+
+use multihash::MultihashGeneric;
+
+pub const TEST_VM_RAND_ARRAY: [u8; 32] = [
+    1u8, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25,
+    26, 27, 28, 29, 30, 31, 32,
+];
+
+// TODO: extend test configuration to allow dynamic toggling for certain features to be intercepted
+// and the behaviour once intercepted
+// https://github.com/anorth/fvm-workbench/issues/9
+#[derive(Debug, Clone)]
+pub struct TestConfiguration {
+    circ_supply: TokenAmount,
+    price_list: PriceList,
+}
+
+pub type Result<T> = std::result::Result<T, ExecutionError>;
+
+pub struct BenchKernel<C: CallManager> {
+    inner_kernel: DefaultKernel<C>,
+    test_config: TestConfiguration,
+}
+
+impl<C> Kernel for BenchKernel<C>
+where
+    C: CallManager,
+{
+    type CallManager = C;
+
+    fn into_inner(self) -> (Self::CallManager, BlockRegistry)
+    where
+        Self: Sized,
+    {
+        self.inner_kernel.into_inner()
+    }
+
+    fn new(
+        mgr: Self::CallManager,
+        blocks: BlockRegistry,
+        caller: ActorID,
+        actor_id: ActorID,
+        method: MethodNum,
+        value_received: TokenAmount,
+        read_only: bool,
+    ) -> Self
+    where
+        Self: Sized,
+    {
+        let default_kernel =
+            DefaultKernel::new(mgr, blocks, caller, actor_id, method, value_received, read_only);
+        let price_list = default_kernel.price_list().clone();
```

Comment:
Can we look up this price list in `self.inner_kernel` on the fly rather than store it in configuration (since we're not overriding any values)?

Similarly for the circulating supply. Hardcoding TOTAL_FILECOIN is going to cause problems one day.